pkg/kubelet/types/pod_update_test.go

```diff
@@ -356,3 +357,53 @@ func TestIsCriticalPodBasedOnPriority(t *testing.T) {
 		})
 	}
 }
+
+func TestIsNodeCriticalPod(t *testing.T) {
+	tests := []struct {
+		name     string
+		pod      *v1.Pod
+		expected bool
+	}{
+		{
+			name:     "critical pod with file source and systemNodeCritical",
+			pod:      getTestPod(configSourceAnnotation(FileSource), nil, scheduling.SystemNodeCritical),
+			expected: true,
+		},
+		{
+			name:     "critical pod with mirror annotation and systemNodeCritical",
+			pod:      getTestPod(configMirrorAnnotation(), nil, scheduling.SystemNodeCritical),
+			expected: true,
+		},
+		{
+			name:     "critical pod using system priority and systemNodeCritical",
+			pod:      getTestPod(nil, &systemPriority, scheduling.SystemNodeCritical),
+			expected: true,
+		},
+		{
+			name:     "critical pod using greater than system priority and systemNodeCritical",
+			pod:      getTestPod(nil, &systemPriorityUpper, scheduling.SystemNodeCritical),
+			expected: true,
+		},
+		{
+			name:     "not a critical pod with api server annotation and systemNodeCritical",
+			pod:      getTestPod(configSourceAnnotation(ApiserverSource), nil, scheduling.SystemNodeCritical),
+			expected: false,
+		},
+		{
+			name:     "not critical if not static, mirror or without a priority and systemNodeCritical",
+			pod:      getTestPod(nil, nil, scheduling.SystemNodeCritical),
+			expected: false,
+		},
+		{
+			name:     "not critical if not static, mirror or without a priority",
+			pod:      getTestPod(nil, nil, ""),
+			expected: false,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			isNodeCriticalPod := IsNodeCriticalPod(test.pod)
+			assert.Equal(t, test.expected, isNodeCriticalPod)
```

Comment:
```suggestion
			require.Equal(t, test.expected, isNodeCriticalPod)
```

(assert has the downside of not terminating the test, so errors stack and become hard to read/triage)